### PR TITLE
twelve app with sentry, buildinfo and update deps with depdencies injection

### PR DESCRIPTION
**Pull request by GreyXor**
- Version generated from VCS tag
- New build struct with version, user agent and compiler infos
- New dependency injection with logger and config
- New Sentry client with high granularity reporting
- New debug mode more verbose
- XDG standards path for config and cache
- No more config file in VCS + Default values are handled in codebase
- Env Var dynamic and prioritised
- New config to set log level
- new error lib that handle sentry
- Server sub-command management and dynamic flag like --help and --version

### CODE REVIEW

File: internal/devtoolkit/devtoolkit.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Package devtoolkit gathers several useful dependencies in different places in the code and avoids making global variables. This is a dependency injection.
+package devtoolkit
+
+import (
+	"github.com/morphysm/famed-github-backend/internal/config"
```
Comment (KilianKae) [inline internal/devtoolkit/devtoolkit.go:5]: Please sort imports.

File: internal/devtoolkit/devtoolkit.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	if err != nil {
+		return nil, eris.Wrap(err, "failed to instantiate build information")
+	}
+
+	toolkit.UserDirs, err = userdirs.NewUserDirs(buildinfo.ProgramName)
+	if err != nil {
+		return nil, eris.Wrap(err, "failed to instantiate user directories")
+	}
+
+	// New configuration, based on environment variables and file
```
Comment (KilianKae) [inline internal/devtoolkit/devtoolkit.go:57]: `and configuration files` ?

File: internal/devtoolkit/devtoolkit.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	toolkit.Config, err = config.Load()
+	if err != nil {
+		return nil, eris.Wrap(err, "failed to load configuration")
+	}
+
+	// Creating a new sentry client and checking the DSN
+	// TODO: CHANGE THE DSN!
+	toolkit.SentryClient, err = sentry.NewClient(sentry.ClientOptions{
+		Dsn:              "https://foo@bar.ingest.sentry.io/TEST",
```
Comment (KilianKae) [inline internal/devtoolkit/devtoolkit.go:66]: I think this should be passed from the config?

Comment (GreyXor) [reply internal/devtoolkit/devtoolkit.go:66]: Love the idea! I will make the change when the other PR config is merged

File: main.go [hunk 4]
```diff
@@ -2,47 +2,78 @@ package main
...
-	if err != nil {
-		log.Panic().Err(err).Msg("failed to load configuration")
-	}
-
-	// Instantiate the server
-	backendServer, err := server.NewServer(cfg)
-	if err != nil {
-		log.Panic().Err(err).Msg("failed to instantiate server")
+	// Check and run server subcommand
+	if arguments.Server != nil {
```
[the remark this answers is not in the document]
Comment (GreyXor) [reply main.go:72]: I changed this

File: main.go [hunk 5]
```diff
@@ -53,6 +55,9 @@ func main() {
 		log.Panic().Err(err).Msg("failed to setup essential components")
 	}
 
+	// be sure to flush Sentry
```
Comment (KilianKae) [inline main.go:58]: Nitpick: `// Flush Sentry on program end/shutdown (or chose one you find more appropriate)`

File: main.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (KilianKae) [reply main.go:72]: Fine by me, I think if more arguments are passed you can probably move the code in the if block to a function.

**Review by KilianKae — CHANGES_REQUESTED** [review]

**Review by KilianKae — CHANGES_REQUESTED** [review]

File: main.go [hunk 6]
```diff
@@ -2,47 +2,82 @@ package main
...
 func main() {
+	// Define default arguments
+	arguments := &Arguments{
+		Server: nil,
+	}
+
+	// Parse the arguments and set server as default sub-command.
+	if arg.MustParse(arguments).Subcommand() == nil {
+		arguments.Server = &Server{}
+	}
```
[the remark this answers is not in the document]
Comment (GreyXor) [reply main.go:50]: --version or --help is handle by `arg.MustParse(arguments)` and MustParse will osExit(0);

**Review by KilianKae — APPROVED** [review]

**Pull request merged**